src/libstore/local-store.cc

```diff
@@ -1076,6 +1076,10 @@ void LocalStore::querySubstitutablePathInfos(const StorePathCAMap & paths, Subst
     if (!settings.useSubstitutes) return;
     for (auto & sub : getDefaultSubstituters()) {
         for (auto & path : paths) {
+            if (infos.find(path.first) != infos.end())
+                // choose first succeeding substituter
+                continue;
```

Comment:
How will this affect signature verification? If I have a substituter that I do not trust in the list of substituters it will only ever yield the path info from that substituter?

Reply:
Again, there is no semantic change here because the info returned from this function is never actually inspected, and returning the first succeeding substituter is at the very least no less, and arguably more, correct than returning the last one, which is is the current behavior. Signature validation is part of the actual substitution, not the initial querying of substituters that uses this function.

Reply:
Thanks for clarifying. I didn't realize that.